src/index.js

```diff
@@ -173,20 +175,32 @@ const singleUserHandler = async (
       process.env.IMAGE_MESSAGE_ENABLED === 'TRUE'
     ) {
       const context = (await redis.get(userId)) || {};
-
+      if (imageProcessingCount >= (process.env.MAX_IMAGE_PROCESS_NUMBER || 3)) {
+        console.log('[LOG] request abort, too many images are processing now.');
+        lineClient('/message/reply', {
+          replyToken,
+          messages: messageBotIsBusy,
+        });
+        clearTimeout(timerId);
+        return;
+      }
+      imageProcessingCount++;
       const res = await downloadFile(otherFields.message.id);
       uploadImageFile(res.clone(), otherFields.message.id);
       await saveImageFile(res, otherFields.message.id);
       const text = await processImage(otherFields.message.id);
-      result = await processText(
-        result,
-        context,
-        type,
-        text,
-        otherFields,
-        userId,
-        req
-      );
+      imageProcessingCount--;
```

Comment:
If anything between `imageProcessingCount++` and `imageProcessingCount--` throws exception, `imageProcessingCount--` will never be executed.

I suggest we do this to make sure `imageProcessingCount--` is always called:

```
imageProcessingCount++;
try {
  // Anything between ++ and --
} finally {
  imageProcessingCount--;
}  
```

Reply:
Updated, thanks for suggestion!!